dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -187,28 +186,19 @@ public <T extends Workflow> Optional<T> findByPath(String path, boolean findPubl
     }
 
     /**
-     * Find if a path already exists in the BioWorkflow or Apptool table since we do not want duplicate names between them.
-     * If creating an apptool, check the workflow table and vice versa.
-     *
+     * Find if a path already exists in the BioWorkflow, Apptool, or Notebook tables, since we do not want duplicate names between them.
      * @param path
-     * @param clazz the table you want to check for a duplicate for
      * @return
      */
-    public <T extends Workflow> void checkForDuplicateAcrossTables(String path, Class<T> clazz) {
-        final List<Workflow> workflows = findByPath(path, false);
-        final List<Workflow> filteredWorkflows = workflows.stream()
-            .filter(workflow -> workflow.getClass() == BioWorkflow.class || workflow.getClass() == AppTool.class)
-            .toList();
-
-        if (filteredWorkflows.size() > 0) {
-            String workflowType;
-            if (clazz == AppTool.class) {
-                workflowType = "tool";
-            } else {
-                workflowType = "workflow";
-            }
-            throw new CustomWebApplicationException("A " + workflowType + " with the same path already exists. Add the 'name' field to the entry you are currently trying to register to give it a unique path.", HttpStatus.SC_INTERNAL_SERVER_ERROR);
-        }
+    public void checkForDuplicateAcrossTables(String path) {
+        final Optional<Workflow> existing = findByPath(path, false).stream()
+            .filter(workflow -> workflow.getClass() != Service.class)
+            .findFirst();
+
+        existing.ifPresent(e -> {
+            String term = e.getEntryType().getTerm();
+            throw new CustomWebApplicationException("A " + term + " with the same path already exists. Add the 'name' field to the entry you are currently trying to register to give it a unique path.", HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
It was like this, but shouldn't this be a 4xx, not a 5xx? It's a bad request, creating a duplicate name...

Reply:
> It was like this, but shouldn't this be a 4xx, not a 5xx? It's a bad request, creating a duplicate name...

Probably true, changed it to `SC_BAD_REQUEST`.